src/helpers/util.js

```diff
@@ -85,6 +85,20 @@ module.exports = {
     return index >= 0 ? array[index] : null
   },
 
+  /*
+    Binary Search:
+    Assumes that @arg array is sorted increasingly
+    return (saying middle=(array[i] + array[i + 1]) / 2 <= target) return i if target < middle; return i + 1 if target >= middle; return array.length - 1 if index > array.length; return null if array[0] > target || array is empty
```

Comment:
Return `i` such that `|array[i] - target|` is smallest among all `i` and `-1` for an empty array. Returns the smallest `i` for multiple candidates.